STRUCTURING_YOUR_PROJECT.md

```diff
@@ -1,5 +1,161 @@
 # Structuring Your Project
 
+This page outlines good practices for Python based projects, but many of the principles discussed here can be applied to 
+other software projects as well.
+
+## Start from a common skeleton which separates key package components
+
+Packages should rely on a common file structure which separates the various elements of a package into separate 
+locations. Remember that a package is not just the code needed to carry out science, but also documentation and 
+deployment. We recommend the following skeletal file structure for some package `examplepackage`:
+ 
+```
+devtools/
+  README.md
+  travis-ci/
+    install_miniconda.sh
+  appveyor-ci/
+    run_with_env.cmd
+  conda-recipe/
+    meta.yaml
+    build.sh
+    bld.bat
+    README.md
+docs/
+examplepackage/
+  tests/
+    __init__.py
+    test_examplepackage.py
+  __init__.py
+  examplepackage.py
+LICENSE.md
+README.md
+setup.py
+.travis.yml
+.appveyor.yml
+.gitignore
+```
+
+This list may appear daunting at first, but each element is either a critical component of what will make the software 
+easy to access, and easy to maintain. The following sections will break down each element in the skeleton and explain 
+why they are helpful. Your package may not need *every* element of this skeleton, but should include many of them.
+
+## The root directory: Primary setup file, CI control, and Git(Hub) Helpers  
+
+```
+LICENSE.md
+README.md
+setup.py
+.travis.yml
+.appveyor.yml
+.gitignore
+```
+
+The root directory should contain the primary `setup.py` file for the `examplepackage`. This is the primary file that 
```

Comment:
Would be good to hyperlink things like `setup.py` to more information about what these files contain.